homeworks/08-js-advanced-1/index.js

```diff
@@ -0,0 +1,129 @@
+// 1. Pluck
+
+function pluck(user, str) {
+  if (!user) return null;
+  if (!str) return user;
+  let coder = str.split(".");
+
+  let value = user;
+
+  for (let code of coder) {
+    if (value[code] !== undefined) {
+      value = value[code];
+    } else {
+      return null;
+    }
+  }
+
+  return value;
+}
+
+// 2. Deep Clone
+
+function clone(obj) {
```

Comment:
The task was: "Create custom deep clone function."
structuredClone - is not custom.